### PR TITLE
Using both struqture 1.0 and 2.0 in noise models

### PR DESCRIPTION
**Pull request by kbarkhqs**

### CODE REVIEW

File: qoqo/python_tests/test_circuit.py [hunk 1]
```diff
@@ -45,5 +46,17 @@ def test_circuit(measurement):
...
+#     hamiltonian = struqture_py.spins.SpinHamiltonianSystem()
+#     hamiltonian.add_operator_product(struqture_py.spins.PauliProduct().z(0).x(2), 1.0)
+#     hamiltonian.add_operator_product("1Z", 2.0)
+
+#     circuit = Circuit()
+#     circuit += ops.ApplyConstantSpinHamiltonian(hamiltonian, 5.0)
+#     circuit += ops.ApplyTimeDependentSpinHamiltonian(hamiltonian, [2.0], {"omega": [1.0]})
+#     assert len(circuit) > 0
+
+
```
Comment (mlodi-hqs) [inline qoqo/python_tests/test_circuit.py:60]: This is here for future reference I guess?

File: qoqo/src/noise_models/continuous_decoherence.rs [hunk 2]
```diff
@@ -56,10 +56,16 @@ impl ContinuousDecoherenceModelWrapper {
...
-            let noise_operator =
-                struqture_py::spins::PlusMinusLindbladNoiseOperatorWrapper::from_pyany(
+            let noise_operator: struqture::spins::PlusMinusLindbladNoiseOperator =
+                match struqture_py::spins::PlusMinusLindbladNoiseOperatorWrapper::from_pyany(
                     lindblad_operator,
-                )?;
+                ) {
+                    Ok(x) => x,
+                    Err(_) => match struqture_py::spins::PlusMinusLindbladNoiseOperatorWrapper::from_struqture_2(lindblad_operator) {
+                        Ok(x) => x.internal,
```
Comment (mlodi-hqs) [inline qoqo/src/noise_models/continuous_decoherence.rs:65]: More of a struqture question: why does the first `Ok` lead to x but the second one to `x.internal`? A 2.0 refactor?

**Review by mlodi-hqs — APPROVED** [review]

File: qoqo/src/noise_models/continuous_decoherence.rs (hunk 2 above)
Comment (kbarkhqs) [reply qoqo/src/noise_models/continuous_decoherence.rs:65]: The first one returns the internal object, but the second function returns a wrapper - this is a good point, I'll add this to the list of things to review for struqture

File: qoqo/python_tests/test_circuit.py (hunk 1 above)
Comment (kbarkhqs) [reply qoqo/python_tests/test_circuit.py:60]: Ah no I have removed this, this test is only meaningful with a manual install of struqture 2.0, which isn't possible in the pipeline

**Pull request merged**